### DISCUSSION TITLE
[Prettier Plugin] Tailwind classes are not being sortet propperly in template strings

### DISCUSSION THREAD

**Discussion by Oshimani** [tailwindlabs/tailwindcss#14974] [Help] [answered]
Hi,

I am currently trying to setup prettier-plugin-tailwind v0.6.8 to sort my tailwind classes within variables and reactive svelte statements. I am using tagged template strings to detect variables with tailwind classes, which works fine.

My problem is that when use string interpolation to update a tailwind class dynamically the sorting does no longer work as expected.

Example:
```typescript
// this works fine
const myClass= tw`cursor-pointer py-1 px-1`

// notice how px-1 and py-1 are separated by the interpolation. They will not be sorted
const myOtherClass = tw`py-1 bg-${color}-600 px-1`;

// all classes in one section will be sorted as expected but only within this section
const myThirdClass = tw`bg-${color}-600 py-1 px-1`; 
```

So the observed behavior is:
When there is no interpolation within a template string, then the sorting works great.
When tailwind classes are separated by a interpolation within a template string, then the sorting does only work within the segments between the individual interpolations.

By interpolation I mean the ${myVar} part of a template string.

I used the following .prettierrc configuration:

```json
{
	// ...
	"plugins": ["prettier-plugin-svelte", "prettier-plugin-tailwindcss"],
	"tailwindFunctions": ["tw", "twMerge", "twJoin"]
}
```

This is my tw function:
```typescript
tw = (strings: TemplateStringsArray, ...values: string[]) => String.raw({ raw: strings }, ...values);
```

Is this expected behavior, or have I configured something wrong?

Thx for your work!

**Comment by philipp-spiess** [comment, marked as answer]
Hey! Unfortunately there is no way to statically analyze a string like this, neither for the Tailwind CSS compiler (which might not even include the `bg-*` classes from this since it doesn't know what `color` resolves to at runtime) nor the prettier plugin.

There's a section about this in the docs as well: https://tailwindcss.com/docs/content-configuration#dynamic-class-names
